src/core/server/http/router/request.ts

```diff
@@ -77,16 +77,113 @@ export interface KibanaRequestEvents {
   completed$: Observable<void>;
 }
 
+/**
+ * Auth status for this request.
+ * @public
+ */
+export interface KibanaRequestAuth {
+  /** true if the request has been successfully authenticated, false otherwise. */
+  isAuthenticated: boolean;
+}
+
 /**
  * Kibana specific abstraction for an incoming request.
  * @public
  */
-export class KibanaRequest<
+export interface KibanaRequest<
   Params = unknown,
   Query = unknown,
   Body = unknown,
   Method extends RouteMethod = any
 > {
+  /**
+   * A identifier to identify this request.
+   *
+   * @remarks
+   * Depending on the user's configuration, this value may be sourced from the
+   * incoming request's `X-Opaque-Id` header which is not guaranteed to be unique
+   * per request.
+   */
+  readonly id: string;
+
+  /**
+   * A UUID to identify this request.
+   *
+   * @remarks
+   * This value is NOT sourced from the incoming request's `X-Opaque-Id` header. it
+   * is always a UUID uniquely identifying the request.
+   */
+  readonly uuid: string;
+
+  /** a WHATWG URL standard object. */
+  readonly url: URL;
+
+  /** matched route details */
+  readonly route: RecursiveReadonly<KibanaRequestRoute<Method>>;
+
+  /**
+   * Readonly copy of incoming request headers.
+   * @remarks
+   * This property will contain a `filtered` copy of request headers.
+   */
+  readonly headers: Headers;
+
+  /**
+   * Whether or not the request is a "system request" rather than an application-level request.
+   * Can be set on the client using the `HttpFetchOptions#asSystemRequest` option.
+   */
+  readonly isSystemRequest: boolean;
+
+  /**
+   * The socket associated with this request.
+   * See {@link IKibanaSocket}.
+   */
+  readonly socket: IKibanaSocket;
+
+  /**
+   * Allow to listen to events bound to this request.
+   * See {@link KibanaRequestEvents}.
+   */
+  readonly events: KibanaRequestEvents;
+
+  /**
+   * The auth status of this request.
+   * See {@link KibanaRequestAuth}.
+   */
+  readonly auth: KibanaRequestAuth;
+
+  /**
+   * URL rewritten in onPreRouting request interceptor.
+   */
+  readonly rewrittenUrl?: URL;
+
+  /**
+   * The path parameter of this request.
+   */
+  readonly params: Params;
+
+  /**
+   * The query parameter of this request.
+   */
+  readonly query: Query;
+
+  /**
+   * The body payload of this request.
+   */
+  readonly body: Body;
+}
+
+/**
+ * Core internal implementation of {@link IKibanaRequest}
+ * @internal
+ */
+export class CoreKibanaRequest<
```

Comment:
One nit: I know we are exporting this for plugins that are using `.from`, but it is marked as `@internal` here. Maybe we could clarify the situation in the comments? Just so in the future we remember that there are actually direct consumers of this class, even if the original intent was for it to be internal.

In addition, perhaps we could have a warning for downstream consumers like "you are probably looking for `KibanaRequest`; do not use this unless you know what you are doing."

---

Also, I think `IKibanaRequest` was a typo here?

```suggestion
/**
 * Core internal implementation of {@link KibanaRequest}
 * @internal
 */
export class CoreKibanaRequest<
```